docs/source/worker.rst

```diff
@@ -218,22 +225,159 @@ usage then the worker will start dumping unused data to disk, even if internal
+    in the 'unmanaged' memory measure above. Ideally, this memory should be for the most
+    part a temporary spike caused by tasks' heap use plus soon-to-be garbage collected
+    objects.
+
+    The time it takes for unmanaged memory to transition away from its "recent" state
+    can be tweaked through the ``distributed.worker.memory.recent-to-old-time`` key in
+    the ``~/.config/dask/distributed.yaml`` file. If your tasks typically run for longer
+    than 30 seconds, it's recommended that you increase this setting accordingly.
+
+    By default, :meth:`distributed.Client.rebalance` and
+    :meth:`distributed.scheduler.Scheduler.rebalance` ignore unmanaged recent memory.
+    This behaviour can also be tweaked using the dask config - see the methods'
+    documentation.
+
+spilled
+    managed memory that has been spilled to disk. This is not included in the 'managed'
+    measure above.
+
+The sum of managed + unmanaged + unmanaged recent is equal definition to the process
+memory.
+
+
+Tweaking OS memory allocation
+-----------------------------
+Different OSs use different default policies when it comes to allocating - and, most
+importantly, freeing - memory. Both the Linux and MacOS memory allocators try to avoid
+performing a kernel call every time the application calls ``free()`` by implementing a
+user-space memory management system. Upon ``free()``, memory can remain allocated in
+user space and potentially reusable by the next ``malloc()`` - which in turn won't
+require a system call either. This is generally very desirable for C/C++ applications
+which have no memory allocator of their own, as it can drastically boost performance at
+the cost of a larger memory footprint. CPython however adds its own memory allocator on
+top, which reduces the need for this additional abstraction (with caveats).
+
+Dask.distributed measures the total process memory (also known as RSS) and uses it for
+decision-making in several points; namely:
+
+- the spill, pause, and terminate thresholds (see above)
+- :meth:`distributed.Client.rebalance` and
+  :meth:`distributed.scheduler.Scheduler.rebalance`
+
+In some cases, it is easy to have these heuristics misbehave because there are large
+amounts of process memory that the libc did not release to the OS yet - but Dask does
+not have the means to tell them apart from a memory leak. This phenomenon is the first
+thing you should investigate when you see large amounts of unexplained unmanaged memory
+in the dashboard or in the logs.
+
+Manually trim memory
+~~~~~~~~~~~~~~~~~~~~
+*Linux only*
+
+It is possible to forcefully release allocated but unutilized memory as follows:
+
+.. code-block:: python
+
+    import ctypes
+
+    def trim_memory() -> int:
+        libc = ctypes.CDLL("libc.so.6")
+        return libc.malloc_trim(0)
+
+    client.run(trim_memory)
+
+This should be only used as a one-off debugging experiment.
```

Comment:
```suggestion
This should be only used as a one-off debugging experiment. Watch the dashboard while running the above code. If unmanaged worker memory (on the "Bytes stored" plot) decreases significantly after calling ``client.run(trim_memory)``, then move on to the next section. Otherwise, you likely do have a memory leak.

Note that you should only run this ``malloc_trim`` if you are using the default glibc memory allocator. When using a custom allocator, this could cause unexpected behavior including segfaults. (If you don't know what this means, you're probably using the default glibc allocator and are safe to run this.)
```